lib/Chunk.js

```diff
@@ -133,9 +138,72 @@ class Chunk {
 		});
 	}
 
+	setModules(modules) {
+		this._modules = new Set(modules);
+		this._modulesIsSorted = false;
+	}
+
+	getNumberOfModules() {
+		return this._modules.size;
+	}
+
+	get modulesIterable() {
+		return this._modules;
+	}
+
+	forEachModule(fn) {
+		this._modules.forEach(fn);
+	}
+
+	mapModules(fn) {
+		const modules = this._modules;
+		const array = new Array(modules.size);
+		let idx = 0;
+		for(let module of modules) {
+			array[idx++] = fn(module, idx, modules);
+		}
+		return array;
+	}
+
+	_ensureModulesSorted() {
+		if(this._modulesIsSorted) return;
+		this._modules = new Set(Array.from(this._modules).sort((a, b) => {
+			if(a.identifier() > b.identifier()) return 1;
+			if(a.identifier() < b.identifier()) return -1;
+			return 0;
+		}));
+		this._modulesIsSorted = true;
+	}
+
+	compareTo(otherChunk) {
+		this._ensureModulesSorted();
+		otherChunk._ensureModulesSorted();
+		if(this._modules.size > otherChunk._modules.size) return -1;
+		if(this._modules.size < otherChunk._modules.size) return 1;
+		const a = this._modules[Symbol.iterator]();
+		const b = otherChunk._modules[Symbol.iterator]();
+		while(true) { // eslint-disable-line
+			const aItem = a.next();
+			const bItem = b.next();
+			if(aItem.done) return 0;
+			const aModuleIdentifier = aItem.value.identifier();
+			const bModuleIdentifier = bItem.value.identifier();
+			if(aModuleIdentifier > bModuleIdentifier) return -1;
+			if(aModuleIdentifier < bModuleIdentifier) return 1;
```

Comment:
These leak open iterators. I wonder how GC works in that case; it might have to execute `finally` blocks...